ugc/ugc_tests/storage_tests.cpp

```diff
@@ -155,107 +160,176 @@ class MwmBuilder
+    auto const railwayId = builder.FeatureIdForRailwayAtPoint(railwayPoint);
+    auto const cafeUGC = MakeTestUGCUpdate(Time(chrono::hours(24 * 10)));
+    auto const railwayUGC = MakeTestUGCUpdate(Time(chrono::hours(24 * 300)));
+
+    {
+      Storage storage(builder.GetIndex());
+      storage.Load();
+      storage.SetUGCUpdate(cafeId, cafeUGC);
+      storage.SetUGCUpdate(railwayId, railwayUGC);
+      storage.SaveIndex();
+    }
+
+    Storage storage(builder.GetIndex());
+    storage.Load();
+    auto const & indexArray = storage.GetIndexesForTesting();
+    TEST_EQUAL(indexArray.size(), 2, ());
+    for (auto const & i : indexArray)
+    {
+      TEST(!i.m_synchronized, ());
+      TEST(!i.m_deleted, ());
+    }
+
+    storage.SetUGCUpdate(cafeId, cafeUGC);
+    TEST_EQUAL(indexArray.size(), 3, ());
+    TEST(DeleteIndexFile(), ());
   }
 
-  Storage storage(builder.GetIndex());
-  storage.Load();
-  auto const & indexArray = storage.GetIndexesForTesting();
-  TEST_EQUAL(indexArray.size(), 2, ());
-  for (auto const & i : indexArray)
+  void ContentTest()
   {
-    TEST(!i.m_synchronized, ());
-    TEST(!i.m_deleted, ());
+    auto & builder = MwmBuilder::Builder();
+    m2::PointD const cafePoint(1.0, 1.0);
+    builder.Build({TestCafe(cafePoint)});
+    auto const cafeId = builder.FeatureIdForCafeAtPoint(cafePoint);
+    auto const oldUGC = MakeTestUGCUpdate(Time(chrono::hours(24 * 10)));
+    auto const newUGC = MakeTestUGCUpdate(Time(chrono::hours(24 * 300)));
+    Storage storage(builder.GetIndex());
+    storage.Load();
+    storage.SetUGCUpdate(cafeId, oldUGC);
+    storage.SetUGCUpdate(cafeId, newUGC);
+    TEST_EQUAL(storage.GetIndexesForTesting().size(), 2, ());
+    auto const toSendActual = storage.GetUGCToSend();
+
+    string data;
+    {
+      using Sink = MemWriter<string>;
+      Sink sink(data);
+      SerializerJson<Sink> ser(sink);
+      ser(newUGC);
+    }
+
+    my::Json ugcNode(data);
+    auto embeddedNode = my::NewJSONObject();
+    ToJSONObject(*embeddedNode.get(), "data_version", cafeId.GetMwmVersion() );
+    ToJSONObject(*embeddedNode.get(), "mwm_name", cafeId.GetMwmName());
+    ToJSONObject(*embeddedNode.get(), "feature_id", cafeId.m_index);
+    ToJSONObject(*ugcNode.get(), "feature", *embeddedNode.release());
+
+    auto array = my::NewJSONArray();
+    json_array_append_new(array.get(), ugcNode.get_deep_copy());
+    auto reviewsNode = my::NewJSONObject();
+    ToJSONObject(*reviewsNode.get(), "reviews", *array.release());
+    unique_ptr<char, JSONFreeDeleter> buffer(json_dumps(reviewsNode.get(), JSON_COMPACT | JSON_ENSURE_ASCII));
+    string const toSendExpected(buffer.get());
+    TEST_EQUAL(toSendActual, toSendExpected, ());
   }
+};
+}  // namespace ugc_tests
 
-  storage.SetUGCUpdate(cafeId, cafeUGC);
-  TEST_EQUAL(indexArray.size(), 3, ());
-  TEST(DeleteIndexFile(), ());
-  TEST(DeleteUGCFile(), ());
+using namespace ugc_tests;
+
+UNIT_CLASS_TEST(StorageTest, Smoke)
```

Comment:
You're doing it wrong. The right way is to get rid of all methods in StorageTests (except dtor), and write them here:
```c++
UNIT_CLASS_TEST(StorageTest, Smoke)
{
  ... code for the smoke here ...
}
```

Reply:
ok :))